CMSgov/onemac
# OY2-16601: base waiver form

Story: https://qmacbis.atlassian.net/browse/OY2-16601
Endpoint: https://d355nq1hyuazr5.cloudfront.net/

### Details

A quick summary of the intended outcome of this PR.

### Changes

- Modify visible "new submission" entries when accessed from Package Dashboard to include a "Base Waiver" option
- Configure a form for (only) base waiver submissions

### Implementation Notes

- Added `legacy` query parameter when accessing New Submission button from Submission Dashboard
- Modified SubmissionForm to accept a Proposed Effective Date
- Modified `submit.js` to support input `type` differing from saved `type`

### Test Plan

1. Log in as `statesubmitteractive`.
2. Click on "New Submission" from the Submission (old) Dashboard, then click "Waiver Action". Verify that there **is not** an entry for "Base Waiver" visible.
3. Click the link to Packages in the nav bar, then click on "New Submission", then "Waiver Action". Verify that there **is** an entry for "Base Waiver" visible.
4. Click on "Base Waiver" and verify the following about the form that appears:
    - The page title is "Base Waiver Submission"
    - There **is no** "Waiver Action" selection present
    - There is an input for "Waiver Authority"
    - There is an input for "Base Waiver Number" and it requires the appropriate format
    - There is an input for "Proposed Effective Date" and it is a date picker widget
    - The attachments section and Additional Information box are consistent with the other forms
5. Fill out the form and submit it. Verify that it successfully submits and that you can see your new base waiver in the dashboards.
